include/aws_jobs.h

```diff
@@ -0,0 +1,239 @@
+ *
+ * @param[in] client_id Client Id of the connected MQTT session.
+ * @param[in] job_id Job Id of the current recived job.
+ */
+#define JOB_ID_UPDATE_TOPIC_TEMPLATE AWS "%s/jobs/%s/update"
+#define JOB_ID_UPDATE_TOPIC_MAX_LEN (AWS_LEN +\
+				     CONFIG_CLIENT_ID_MAX_LEN +\
+				     JOB_ID_MAX_LEN +\
+				     14)
+
+
+/** @brief Job Execution Status. */
+enum execution_status {
+	QUEUED = 0,
+	IN_PROGRESS,
+	SUCCEEDED,
+	FAILED,
+	TIMED_OUT,
+	REJECTED,
+	REMOVED,
+	CANCELED
+};
+
+
+/** @brief Mapping of enum to strings for Job Execution Status. */
+const char *execution_status_strings[] = {
+	[QUEUED] = "QUEUED",
+	[IN_PROGRESS] = "IN_PROGRESS",
+	[SUCCEEDED] = "SUCCEEDED",
+	[FAILED] = "FAILED",
+	[TIMED_OUT] = "TIMED_OUT",
+	[REJECTED] = "REJECTED",
+	[REMOVED] = "REMOVED",
+	[CANCELED] = "CANCELED"
+};
+
+
+/**
+ * @brief Constructs the get topic for accepted/rejected AWS Jobs and subscribe
+ *	to it.
+ *
+ * @param[in] client Connected MQTT client instance.
+ *
+ * @retval 0 If successful otherwise the return code of mqtt_subscribe or
+ *	snprintf.
+ */
+int aws_jobs_subscribe_get(struct mqtt_client *client);
+
+
+/**
+ * @brief Constructs then the notify-next topic subscribes for reciving
+ *	AWS IoT jobs.
+ *
+ * @param[in] client Connected MQTT client instance.
+ *
+ * @retval 0 If successful otherwise the return code of mqtt_subscribe or
+ *	snprintf.
+ */
+int aws_jobs_subscribe_notify_next(struct mqtt_client *client);
+
+
+/**
+ * @brief Constructs then subscribes to notify topic for reciving  AWS IoT jobs.
+ *
+ * @param[in] client Connected MQTT client instance.
+ *
+ * @retval 0 If successful otherwise the return code of mqtt_subscribe or
+ *	snprintf.
+ */
+int aws_jobs_subscribe_notify(struct mqtt_client *client);
+
+
+/**@brief Subscribe to the expected topics of the AWS documentation
+ * https://docs.aws.amazon.com/iot/latest/developerguide/jobs-devices.html.
+ *
+ * @param[in] client Connected MQTT client instance.
+ * @param[in] notify_next Subscribes to notify next if true and notify if false.
+ *
+ * @retval 0 If successful otherwise the return code of mqtt_subscribe or
+ *	snprintf.
+ */
+int aws_jobs_subscribe_expected_topics(struct mqtt_client *client,
+				       bool notify_next);
+
+
+/**
+ * @brief Constructs the get topic for accepted and rejeceted AWS jobs.
+ *
+ * @param[in] client Connected MQTT client instance.
+ * @param[in] job_id JobId of the current accepted job.
+ *
+ * @retval 0 If successful otherwise the return code of mqtt_subscribe or
+ *	snprintf.
+ */
+int subscribe_job_id_get_topic(struct mqtt_client *client, u8_t *job_id);
+
+
+/**
+ * @brief AWS jobs update job execution status function.
+ *
+ * This implements the minimal requirements for doing an AWS job only
+ * updating status and status details update are supported.
+ *
+ * @param[in] client Initialized and connected MQTT Client instance.
+ * @param[in] job_id The ID of the job which you are updating. This will be a
```

Comment:
A bit more about strings :), here's my proposal:
Please make it clear (in docs) when strings are expected to be
 1. UTF-8
 2. null-terminated.

When they are expected to be ASCII, please use `char*`.

Reply:
How about making it easier and say all string give to the library is in utf-8 format. As this is what is expected from MQTT and JSON.

Reply:
Sure, sounds good. Remember to still mark args as strings in the docs when this is not obvious.